### ISSUE TITLE
when material-ui's css-in-js solution is ready, deprecate emotion and jss

### ISSUE THREAD

**Issue by bcolloran** [iodide-project/iodide#1527]
see #1524

**Comment by hamilton** [comment]
I could be misunderstanding, but "deprecate emotion" would mean writing all of our components in `material-ui`'s css-in-js solution, which is most assuredly not going to be as mature as any of the other css-in-js libraries. That seems like a lot of work for very little actual gain to me - perhaps a lot more work than just deprecating material-ui. 

In the long run I think we'd be more interested in deprecating material ui on the whole, correct? We use emotion far, far more often than we use material-ui, and we're likely to use it more in the future. As of now, I think we're only using material-ui for a few things in the client ("task buttons" in the menu, icons, the menu on the left that we're considering deprecating), and almost nothing on the server.

Could you clarify a bit further?

**Comment by bcolloran** [comment]
I don't think we want to deprecate material-ui, using an off the shelf widget + theming library buys us a lot in terms of consistency and simplicity. Hopefully we'll soon have more people building more stuff, and it'll be nice for people to have a robust and well documented palette of widgets to choose from that will result in a consistent look (as in #1507, where will was able to just grab the list widget from material to get going with little overhead).

I'm curious to hear your thoughts about css-in-js in general though -- as i mention in #1524, i'm not sure what problem it's really solving for us that inline styles + THEME couldn't solve, except for in case of things like :hover selectors. For those few cases, i expect a less mature solution would work. but if there are other problems that emotion is solving for us, i'm interested to learn -- i may still have an incomplete understanding of the problem space.

**Comment by hamilton** [comment]
Agreed that we don't want to deprecate material-ui (that's a lot of work and things are mostly fine), but my point was that deprecating emotion (which we use for lots of components that don't exist in material-ui) would actually be quite a bit of work as well, so I just want to make sure whatever we're replacing emotion with is mature enough. I am hesitant to put all my eggs in the material-ui basket. 

What motivated the issue in the first place was me using emotion to style a material ui component. We haven't encountered this issue before because we haven't used emotion to style material ui stuff. And for good reason - I primarily have used emotion to style plain html elements + other emotion components (buttons, lists, containers, etc.), and aside from the high-quality icons, never ever touch material-ui anymore. In fact, what caused us to discover this issue was me styling material-ui icons w/ emotion!

I will have to look a bit more about the benefits of inline styles, but I do think that (for me) writing actual css is a huge benefit since I'm very comfortable writing css and can enjoy the best of both worlds here, whereas you _can't_ really get that from inline css as it is currently practiced.

At any rate, I'll spend a bit more time around this + read your posts and develop a more nuanced worldview, and we can take it from there. In the short term I think the PR that motivated this issue will keep us going as-is for now.

**Comment by bcolloran** [comment]
cool, that all makes sense.

let me know if you find any illuminating articles -- i'm interested in learning more about this stuff and continuing to develop the worldview with you
